pkg/auth/cloud/azure/env.go

```diff
@@ -0,0 +1,147 @@
+package azure
+
+import (
+	log "github.com/cloudposse/atmos/pkg/logger"
+	"github.com/cloudposse/atmos/pkg/perf"
+)
+
+// problematicAzureEnvVars lists environment variables that should be cleared by Atmos auth
+// to avoid authentication conflicts when using Azure SDK.
+//
+// These variables can interfere with Atmos's Azure authentication flow.
+// By clearing these variables before authentication, we ensure Atmos uses only its managed
+// credentials.
+var problematicAzureEnvVars = []string{
+	// Authentication credentials.
+	"AZURE_CLIENT_ID",
+	"AZURE_CLIENT_SECRET",
+	"AZURE_CLIENT_CERTIFICATE_PATH",
+	"AZURE_USERNAME",
+	"AZURE_PASSWORD",
+	"AZURE_FEDERATED_TOKEN_FILE",
+
+	// Subscription/tenant configuration.
+	"AZURE_SUBSCRIPTION_ID",
+	"AZURE_TENANT_ID",
+
+	// Location configuration.
+	"AZURE_LOCATION",
+	"AZURE_REGION",
+
+	// Config file paths.
+	"AZURE_CONFIG_DIR",
+
+	// Terraform/ARM specific vars.
+	"ARM_CLIENT_ID",
+	"ARM_CLIENT_SECRET",
+	"ARM_CLIENT_CERTIFICATE_PATH",
+	"ARM_SUBSCRIPTION_ID",
+	"ARM_TENANT_ID",
+	"ARM_USE_MSI",
+	"ARM_USE_CLI",
+	"ARM_USE_OIDC",
+}
+
+// conflictingCloudEnvVars lists environment variables from other cloud providers
+// that should be cleared when using Azure authentication to avoid conflicts.
+var conflictingCloudEnvVars = []string{
```

Comment:
Why would we want to clear these? That would prevent using multiple identities, for example a backend on S3 and deployments to Azure. I don't think we should be clearing these.